### ISSUE TITLE
SentencePair Regression

### ISSUE THREAD

**Issue by overlookcurl** [deepset-ai/FARM#688]
**Question**
Hi. Thank you for this wonderful repository. I am still in the early stages of using FARM.
I want to perform a sentencepair regression task with FARM. Basically I have two continues values for each sentence pair. I was going to use two regression heads for that. 

However, I noticed that There is no seperate processor for this. (TextPairRegressionProcessor), So I assumed that I have to use TextPairClassificationProcessor with task_type regression. 

Then I was examining the TextPairClassificationProcessor and noticed that it always checks for label_list. 

![image](https://user-images.githubusercontent.com/74681892/104747149-12d2d180-5748-11eb-9dd1-df6aa138fd75.png)

1. If my approach is right to use TextPairClassificationProcessor, what should I provide for label_list in the regression case?
2. If it is not will you be able to tell me the correct process to do text pair regression task?

Thank you in Advance and Thank you for this wonderful repo again

**Comment by Timoeller** [comment]
Hey @overlookcurl thanks for digging into FARM. You already found the relevant parts - but you cannot directly use them for your use-case. 

So using the textpairprocessor with task_type regression wont work unfortunately.
But you can phrase a regression task as a classification task and bin your regressor variable. I would honestly recommend it, the few experiments we did with the pure regression datasets were not promising. Some people say regression is less stable than using the crossentropy on multiple classes.
Would that be an option? If not you would need to implement an own TextPairRegressionProcessor.

**Comment by overlookcurl** [comment]
Hi @Timoeller, 
Thank you very much for the response. For my dataset when I used huggingface transformer models, it worked fairly well. So I think I will go with implementing TextPairRegressionProcessor. Will open this issue, if I come across any issues when doing that. 

Thank you so much again

**Comment by Timoeller** [comment]
Hey @overlookcurl I actually overlooked something ;) Our TextPairClassificationProcessor actually has some rather hacky way of supporting regression labels. But those labels are not scaled, so they should not work well with a Regressionhead that uses MSE as loss function. Does that make sense to you?

We are refactoring the TextPairClassificationProcessor soon and it might make sense to implement a better support for regression there. See #701 for general improvements to the Processor and deepset-ai/haystack#423 and linked issue for regression support.

Btw. for what use-case do you need the TextPairRegression?

**Comment by overlookcurl** [comment]
Hi @Timoeller, 

Thank you very much. My task is tralation quality estimation task. So basically I have a source sentence, target sentence, an overall quality of the translation between 0-1, and the quality of the each word. 

So I was thinking to use sentencepair regression with the token classifiers in a multi task setting. Was I clear?

**Comment by Timoeller** [comment]
> Was I clear?

Totally, that makes a lot of sense. Using a regression on the translation quality seems reasonable. I guess the challenge will come in combining the losses of overall translation quality and the individual word translation quality as the overall quality should account for much more.

Anyways, I can ping you once the implementation is done, I plan to have it finished by 17th of Feb. If that is too late I would highly recommend binning your overall score into, lets say, 5 bins from 0-1 and use multiclass classification. I know this takes away the relation between classes, class 1 to 5 is as similar to the model as 4 to 5, but in our experience the Crossentropy loss used for multiclass somehow works better that MSE loss for regression.

Lets keep this issue open for now and post updates in here.

**Comment by overlookcurl** [comment]
Thank you very much for the great work
Great, I will wait till you implement it then.
